sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -842,6 +842,142 @@ class FileBasedDataSourceSuite extends QueryTest
       }
     }
   }
+
+  test("SPARK-31116: Select simple parquet with case insensitive mode") {
+    Seq("true", "false").foreach { nestedSchemaPruningEnabled =>
+      withSQLConf(
+        SQLConf.CASE_SENSITIVE.key -> "false",
+        SQLConf.NESTED_SCHEMA_PRUNING_ENABLED.key -> nestedSchemaPruningEnabled) {
+        withTempPath { dir =>
+          val path = dir.getCanonicalPath
+
+          // Prepare values for testing specific parquet record reader
+          Seq("A").toDF("camelCase").write.parquet(path)
+
+          val exactSchema = new StructType().add("camelCase", StringType)
+          checkAnswer(spark.read.schema(exactSchema).parquet(path), Row("A"))
+
+          // In case insensitive manner, parquet's column cases are ignored
```

Comment:
This describes the correct current behavior. Apache Spark doesn't have an issue for non-nested schema.